### PR TITLE
Adding context version of Check and Response.Apply methods.

### PR DESCRIPTION
**Pull request by henvic**
Supersedes #7.

### CODE REVIEW

**Comment by zenorocha** [comment]
Ping

**Comment by henvic** [comment]
Hi @kyleconroy, I see you have done some commits on this project recently. Can you please take a look at this?

I believe it's pretty straight-forward and shouldn't take much time.

Thanks.

**Comment by kyleconroy** [comment]
@henvic yes, can do. I honestly had forgotten this was open. Sorry!

File: sdk_ctx.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+// CheckContext is like Check but includes a context.
+func CheckContext(ctx context.Context, appID string, opts Options) (Response, error) {
+	var req, err = checkRequest(appID, &opts)
+
+	if err != nil {
+		return Response{}, err
+	}
+
+	req = req.WithContext(ctx)
+	return doCheckRequest(opts, req)
```
Comment (kyleconroy) [inline sdk_ctx.go]: Instead of this being two lines, can we just make this `doCheckRequest(opts, req.WithContext(ctx))`?

File: sdk_ctx.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+func (r Response) ApplyContext(ctx context.Context) error {
+	var req, opts, err = r.applyRequest()
+
+	if err != nil {
+		return err
+	}
+
+	req = req.WithContext(ctx)
+
+	return r.applyUpdate(req, opts)
```
Comment (kyleconroy) [inline sdk_ctx.go]: Same comment as above `r.applyUpdate(req.WithContext(ctx), opts)`

File: .travis.yml [hunk 3]
```diff
@@ -2,4 +2,6 @@ language: go
 go:
   - 1.4
   - 1.5
+  - 1.7
+  - 1.9
```
Comment (kyleconroy) [inline .travis.yml]: Can we add 1.10 and 1.11?

**Review by kyleconroy — CHANGES_REQUESTED** [review]
Just a few small changes. Sorry again that this review took so long to happen.

File: .travis.yml (hunk 3 above)
Comment (henvic) [reply .travis.yml]: tip as well? :)

Comment (henvic) [reply .travis.yml]: with matrix: allow_failures :)

File: .travis.yml [hunk 4]
```diff
@@ -2,4 +2,12 @@ language: go
...
+  - 1.10
```
Comment (kyleconroy) [inline .travis.yml]: We'll need to put all these version numbers in quotes. See https://github.com/travis-ci/travis-ci/issues/9247 for more information

**Review by kyleconroy — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 4 above)
Comment (henvic) [reply .travis.yml]: Asking myself: how did I never notice it? Fixed.

**Comment by kyleconroy** [comment]
There appears to be a `go vet` failure in `sdk_test.go` which is unrelated to your changes. Either you'll need to fix it or you'll need to remove `1.10` and `1.11` from the matrix and I can fix the issue in a separate PR.

**Review by kyleconroy — APPROVED** [review]

**Comment by kyleconroy** [comment]
Thanks @henvic. I'll cut a new release that includes the new methods

**Pull request merged**